libs/utils/env.py

```diff
@@ -474,19 +485,36 @@ def _init_target(self, force = False):
             # Brackets are there to let the output dir be created automatically
                                     '--workload-automation-vio={}'])
 
+            # Add variant suffix
+            suffix = os.path.splitext(os.path.basename(
+                     system_info['platform']['description']))[0]
+            board_file = self.conf['board'].lower() + suffix + '.json';
+            self._board_path = os.path.join(basepath, 'libs/utils/platforms',
+                                      board_file)
+            if not os.path.exists(self._board_path):
+                self._log.warning('Board descriptor file %s was not found.',
+                                  self._board_path)
+                self._board_path = None
+                board = None
+            else:
+                board = self._load_board()
```

Comment:
No, that's not yet convincing to me... looking again at the configurations settings you are adding for gem5 it seems that your concept of "board" cannot be just "gem5". But instead whatever platform configuration you specify under:
`system_info['platform']['description']`

Thus, why not simply overriding the `board` attribute in the gem5 case?

I mean, here you just set:
```python
suffix = os.path.splitext(os.path.basename(system_info['platform']['description']))[0]
# Redefined board to match the specific Gem5 platform we use
self.conf['board'] = self.conf['board'] + suffix
```
and then:
1. the refactoring below is not required
2. we have `self.conf['board']` properly set to represent a specific target

Could that work?